lazyflow/utility/reorderAxesDecorator.py

```diff
@@ -0,0 +1,447 @@
+               member != 'setupOutputs'  # will be adapted individually
+               ]
+
+    class_methods = [member for member, member_type in cls.__dict__.items()
+                     if isinstance(member_type, classmethod)]
+
+    property_methods = [member for member, member_type in cls.__dict__.items()
+                        if isinstance(member_type, property)]
+
+    # overwrite methods with guarded versions thereof
+    for meth in methods:
+        old_meth = getattr(cls, meth)
+
+        def guard(fn):
+            @functools.wraps(old_meth)
+            def wrap(self, *args, **kwargs):
+                if self._inner_call:
+                    return fn(self, *args, **kwargs)
+                else:
+                    try:
+                        self._inner_call = True
+                        ret = fn(self, *args, **kwargs)
+                        self._inner_call = False
+                        return ret
+                    except Exception:
+                        raise
+                    finally:
+                        self._inner_call = False
+            return wrap
+
+        setattr(cls, meth, guard(old_meth))
+
+    # overwrite class methods with guarded versions thereof
+    for meth in class_methods:
+        old_meth = getattr(cls, meth)
+
+        def guard(fn):
+            @functools.wraps(old_meth)
+            def wrap(self, *args, **kwargs):
+                if self._inner_call:
+                    return fn(*args, **kwargs)
+                else:
+                    try:
+                        self._inner_call = True
+                        ret = fn(*args, **kwargs)
+                        self._inner_call = False
+                        return ret
+                    except Exception:
+                        raise
+                    finally:
+                        self._inner_call = False
+            return wrap
+
+        setattr(cls, meth, guard(old_meth))
+
+    # overwrite property methods with guarded versions thereof
+    for meth in property_methods:
+        old_get = getattr(cls, meth).__get__
+
+        def guard(fn):
+            @property
+            def wrap(self):
+                if self._inner_call:
+                    return fn(self)
+                else:
+                    try:
+                        self._inner_call = True
+                        ret = fn(self)
+                        self._inner_call = False
+                        return ret
+                    except Exception:
+                        raise
+                    finally:
+                        self._inner_call = False
+            return wrap
+
+        setattr(cls, meth, guard(old_get))
+
+    return cls
+
+
+def reorder(cls):
+    """
+        Decorator function to reorder all input and output channels, to preserve an inner axes order within the
+        operator.
+        Must be used in combination with reorder_options!!!
```

Comment:
No. This is explained in the doc string of reorder_options. 
The reason is that a decorator with options has to return a function (here: 'set_options'). When initializing the operator's meta class, the operator class has to have inherited from the operator base class. the type, returned by a decorator with options will be function though... therefore I use the reorder_options decorator to add the options to the class, so that the reorder_decorator gets the class as argument and can return the class. I guess it is just a python implementation detail of decorators, but maybe I am overlooking something.